openmdao/core/tests/test_group.py

```diff
@@ -96,6 +96,146 @@ def configure(self):
 
 class TestGroup(unittest.TestCase):
 
+    def test_promotes_outputs_in_config(self):
+
+        class SimpleGroup(om.Group):
+
+            def setup(self):
+
+                self.add_subsystem('comp1', om.IndepVarComp('x', 5.0))
+                self.add_subsystem('comp2', om.ExecComp('b=2*a'))
+
+            def configure(self):
+                self.promotes('comp2', outputs=['b'])
+
+        top = om.Problem(model=SimpleGroup())
+        top.setup()
+
+        self.assertEqual(top['b'], 1)
+        with self.assertRaises(UnboundLocalError) as cm:
+            top['a']
+
+        self.assertEqual(str(cm.exception),
+                         "local variable 'val' referenced before assignment")
+
+    def test_promotes_inputs_in_config(self):
```

Comment:
I think we still need a test where a subsystem promotes a variable to one alias, and then a parent tries to promote it to a different alias.  That should raise an exception.  Promoting it to the same alias would be ok.